build/modules/SitecoreImageBuilder/1.0.0/Public/Invoke-PackageRestore.ps1

```diff
@@ -69,11 +69,14 @@ function Invoke-PackageRestore
     $expected | ForEach-Object {
         $filePath = $_
 
+        $fileName = ($filePath.Replace(("{0}" -f $destinationPath), "")) -replace "([\/\\])", ""
+        $package = $packages.$fileName
+
         if (Test-Path $filePath -PathType Leaf)
         {
-            $requiredFile = Get-Item -Path $filePath
+            $requiredFileHash = $(Get-FileHash -Path $filePath).Hash
 
-            if ($requiredFile.Length -gt 0)
+            if ($package.hash -eq $requiredFileHash)
```

Comment:
What happens in the event where the url and hash are not provided in the `sitecore-packages.json` file?

Do we need to add the hash for all files in the sitecore-packages.json file?

ie:
``` 
"Data Exchange Framework 5.0.0 rev. 01466.scwdp.zip": {
        "url": "",
        "hash": ""
    }

Reply:
This is a good point. Should I add check to skip hash check if it's empty?
For me it sounds better to put hashes for all files in `sitecore-packages.json`, but I understand that might be tricky to do it.

Reply:
It's up to you. Ideally the hash for all files exists, but it's probably more realistic to skip if hash is empty. Otherwise someone has to download the zip, convert it using SAT and then update the hash.

It's something that can be done over time and perhaps we can enforce future PRs to include the hash for items added to the json (guilty as charged) and as modifications are made, we can add the hash.

I'd say for now just add logic to skip if empty.

Reply:
@jeanfrancoislarente 
I updated my changes: if hash in `sitecore-packages.json` is empty, then old check will be used, otherwise file hash will be verified.